OpenRA.Mods.Common/Effects/Missile.cs

```diff
@@ -673,16 +666,29 @@ int IncreaseAltitude(int predClfDist, int diffClfMslHgt, int relTarHorDist, int
 			{
 				// Aim to attain cruise altitude as soon as possible while having the absolute value
 				// of vertical facing bound by the maximum vertical rate of turn
-				var vDist = new WVec(-diffClfMslHgt - info.CruiseAltitude.Length, -speed, 0);
-				desiredVFacing = (sbyte)vDist.HorizontalLengthSquared != 0 ? vDist.Yaw.Facing : vFacing;
-				desiredVFacing = desiredVFacing.Clamp(-info.VerticalRateOfTurn, info.VerticalRateOfTurn);
-
+				desiredVFacing = CalculateDesiredVFacingForAscent(vFacing, diffClfMslHgt);
 				ChangeSpeed();
 			}
 
 			return desiredVFacing;
 		}
 
+		int CalculateDesiredVFacingForDescent(WVec vDist, int vFacing)
```

Comment:
Those method names are most likely wrong. If anyone has better suggestions, have at it.